### PR TITLE
Add manifest proto and initial CI scripts

### PR DESCRIPTION
**Pull request by liamawhite**
Signed-off-by: Liam White <liam@tetrate.io>

### CODE REVIEW

File: ci/lint [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+SUCCESS=true
+
+# Ensure Licenses are present
+if licenser verify -r .; then
+    echo "Licenses are present on all recognised files."
+else
+    echo "Licenses are missing."
+    SUCCESS=false
```
Comment (taiki45) [inline ci/lint]: nit: I prefer early exits, but since current impl has the advantage that this can show full lint errors, it's okay.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+version: 2.0
+
+jobs:
+  lint:
+    resource_class: small
+    docker:
+      - image: circleci/golang:1.12.6
+    steps:
+      - checkout
+      - run: wget -c https://github.com/liamawhite/licenser/releases/download/v0.4.0/licenser_0.4.0_Linux_x86_64.tar.gz -O - | sudo tar -xz -C /usr/bin
```
Comment (taiki45) [inline .circleci/config.yml]: nit: checksum? the environment is sandboxed though.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+.vscode
```
Comment (taiki45) [inline .gitignore]: nit: I think this is not related to this project, rather related to developers environment, so devs should ignore in their per-user gitignore?

File: ci/lint [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+SUCCESS=true
+
+# Ensure Licenses are present
+if licenser verify -r .; then
+    echo "Licenses are present on all recognised files."
+else
+    echo "Licenses are missing."
```
[the remark this answers is not in the document]
Comment (liamawhite) [reply ci/lint]: The idea here being that you just run `ci/format` and then `git commit...`?
